public/app/plugins/datasource/grafana-azure-monitor-datasource/azure_monitor/azure_monitor_datasource.ts

```diff
@@ -509,6 +500,35 @@ export default class AzureMonitorDatasource extends DataSourceWithBackend<AzureM
       });
   }
 
+  isValidConfig() {
```

Comment:
it strikes me as a bit odd that a truthy result from isValidConfig is actually an invalid config but it looks like we already have that naming convention haha. I also think explicitly returning undefined is sort of strange, but we also already do that so I suppose this is probably not worth changing at the moment. Perhaps something todo later.

Reply:
hmmm I yeah @kostrse I understand this is the same pattern as is elsewhere (unfortunately), but you do you think we could rename the function (just this one here) just to make it a bit clearer, to reduce confusion/temptation with `if (isValidConfig()) return success`?

Reply:
Yes, this is existing implementation and it has even deeper problems than just naming :) There's  [isConfigured](https://github.com/grafana/grafana/blob/4779de9467857b80658574ace2f41ff14bcd25a5/public/app/plugins/datasource/grafana-azure-monitor-datasource/azure_monitor/azure_monitor_datasource.ts#L69-L71) method which is also does validation called in datasource.ts [here](https://github.com/grafana/grafana/blob/4779de9467857b80658574ace2f41ff14bcd25a5/public/app/plugins/datasource/grafana-azure-monitor-datasource/datasource.ts#L145-L155) even before the `isValidConfig` has chance to return validation results. So, in most cases error messages from `isValidConfig` never even returned.

I made some consmetic chagnes, I renamed the method to `validateDatasource`, added return type etc. But validation needs much deeper overhaul.